drivers/ieee802154/ieee802154_gecko.c

```diff
@@ -0,0 +1,588 @@
+/*
+ * Copyright (c) 2020 Piotr Mienkowski
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(ieee802154_gecko, CONFIG_IEEE802154_DRIVER_LOG_LEVEL);
+
+#include <errno.h>
+#include <kernel.h>
+#include <arch/cpu.h>
+#include <debug/stack.h>
+#include <soc.h>
+#include <device.h>
+#include <init.h>
+#include <debug/stack.h>
+#include <net/net_if.h>
+#include <net/net_pkt.h>
+#include <sys/byteorder.h>
+#include <string.h>
+#include <random/rand32.h>
+#include <net/ieee802154_radio.h>
+
+#include <em_system.h>
+#include <rail.h>
+#include <rail_ieee802154.h>
+
+/* Defines maximum frame size used by the radio */
+#define RADIO_MAX_FRAME_SIZE                  128
+/* Maximum time to wait for RAIL to send the packet */
+#define TX_PACKET_SENT_TIMEOUT         K_MSEC(100)
+#define IEEE802154_FCS_LENGTH		        2
+
+struct ieee802154_gecko_context {
+	/* RAIL internal Tx FIFO */
+	u8_t rail_tx_fifo[RADIO_MAX_FRAME_SIZE] __aligned(RAIL_FIFO_ALIGNMENT);
+
+	RAIL_Handle_t rail_handle;
+
+	/* Pointer to the network interface. */
+	struct net_if *iface;
+
+	/* Device 802.15.4 long address. */
+	u8_t mac[8];
+
+	/* TX synchronization semaphore. Unlocked when frame has been sent or
+	 * send procedure failed.
+	 */
+	struct k_sem tx_wait;
+
+	/* TX result, updated in radio transmit callbacks. */
+	int tx_status;
+
+	/* Transmit / receive channel */
+	u16_t channel;
+};
+
+struct ieee802154_gecko_dev_cfg {
+	/* Initial TX power level in RAIL raw units */
+	RAIL_TxPowerLevel_t init_tx_power_level_raw;
+};
+
+#define DEV_NAME(dev) ((dev)->config->name)
+
+#define DEV_CFG(dev) \
+	((const struct ieee802154_gecko_dev_cfg *const)(dev)->config->config_info)
+
+#define DEV_DATA(dev) \
+	((struct ieee802154_gecko_context *const)(dev)->driver_data)
+
+static const RAIL_CsmaConfig_t rail_csma_config =
+	RAIL_CSMA_CONFIG_802_15_4_2003_2p4_GHz_OQPSK_CSMA;
+
+static const RAIL_LbtConfig_t rail_lbt_config = {
+	.lbtMinBoRand = 0,
+	.lbtMaxBoRand = 10,
+	.lbtTries = 5,
+	.lbtThreshold = -75,
+	.lbtBackoff = 320,   /* 20 symbols at 16 us/symbol */
+	.lbtDuration = 128,  /* 8 symbols at 16 us/symbol */
+	.lbtTimeout = 0,     /* No timeout */
+};
+
+static const RAIL_IEEE802154_Config_t rail_ieee802154_config = {
+	.addresses = NULL,
+	.ackConfig = {
+		.enable = false,
+		.ackTimeout = 672,
+		.rxTransitions = {
+			RAIL_RF_STATE_RX,
+			RAIL_RF_STATE_RX,
+		},
+		.txTransitions = {
+			RAIL_RF_STATE_RX,
+			RAIL_RF_STATE_RX,
+		},
+	},
+	.timings = {
+		.idleToRx = 100,
+		.idleToTx = 100,
+		.rxToTx = 192,
+		.txToRx = 192,
+		.rxSearchTimeout = 0,
+		.txToRxSearchTimeout = 0,
+	},
+	.framesMask = RAIL_IEEE802154_ACCEPT_STANDARD_FRAMES,
+	.promiscuousMode = false,
+	.isPanCoordinator = false,
+};
+
```

Comment:
```suggestion
RAIL_DECLARE_TX_POWER_VBAT_CURVES_ALT;
```

Will be used to compute the proper TX power value